Flaky RunOptionsDefaultsComponentTest

Seen this twice on Kokoro today. The offending assertion was added into an existing test in #2375.

```
testEnablement_existingStagingLocation(com.google.cloud.tools.eclipse.dataflow.ui.preferences.RunOptionsDefaultsComponentTest)  Time elapsed: 0.094 sec  <<< FAILURE!
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.google.cloud.tools.eclipse.dataflow.ui.preferences.RunOptionsDefaultsComponentTest.testEnablement_existingStagingLocation(RunOptionsDefaultsComponentTest.java:264)
```

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java
@@ -231,6 +231,7 @@ public class RunOptionsDefaultsComponentTest {
       }
       Thread.sleep(50);
     } while (i++ < 200 && !verifyResult.isDone());
+    component.validate();
     assertTrue(selector.isEnabled());
     assertNotNull(selector.getSelectedCredential());
     assertTrue(projectID.isEnabled());
@@ -256,6 +257,7 @@ public class RunOptionsDefaultsComponentTest {
       }
       Thread.sleep(50);
     } while (i++ < 200 && !fetchResult.isDone() && !verifyResult.isDone());
+    component.validate();
     assertTrue(selector.isEnabled());
     assertNotNull(selector.getSelectedCredential());
     assertTrue(projectID.isEnabled());
@@ -280,6 +282,7 @@ public class RunOptionsDefaultsComponentTest {
       }
       Thread.sleep(50);
     } while(i++ < 200 && !createButton.isEnabled());
+    component.validate();
     assertTrue(verifyResult.isDone());
     assertTrue(selector.isEnabled());
     assertNotNull(selector.getSelectedCredential());

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java
@@ -62,6 +62,7 @@ import org.eclipse.swt.widgets.Button;
 import org.eclipse.swt.widgets.Combo;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
 import org.eclipse.swt.widgets.Label;
 import org.eclipse.swt.widgets.Text;
 import org.eclipse.ui.PlatformUI;
@@ -213,7 +214,9 @@ public class RunOptionsDefaultsComponent {
     validate();
   }
 
-  private void validate() {
+  @VisibleForTesting
+  void validate() {
+    Preconditions.checkState(Display.getCurrent() != null, "Must be called on SWT UI thread");
     // we set pageComplete to the value of `allowIncomplete` if the fields are valid
     setPageComplete(false);
     messageTarget.clear();

